Pcap++/src/DpdkDeviceList.cpp

```diff
@@ -192,11 +183,15 @@ bool DpdkDeviceList::initDpdkDevices(uint32_t mBufPoolSizePerDevice, uint16_t mB
 	// Initialize a DpdkDevice per port
 	for (int i = 0; i < numOfPorts; i++)
 	{
-		DpdkDevice* newDevice = new DpdkDevice(i, mBufPoolSizePerDevice, mBufDataSize);
+		auto newDevice = std::unique_ptr<DpdkDevice>(new DpdkDevice(i, mBufPoolSizePerDevice, mBufDataSize));
 		PCPP_LOG_DEBUG("DpdkDevice #" << i << ": Name='" << newDevice->getDeviceName() << "', PCI-slot='" << newDevice->getPciAddress() << "', PMD='" << newDevice->getPMDName() << "', MAC Addr='" << newDevice->getMacAddress() << "'");
-		m_DpdkDeviceList.push_back(newDevice);
+		m_DeviceList.pushBack(std::move(newDevice));
 	}
 
+	// Full update of all elements of the view vector to synchronize them with the main vector.
+	m_DpdkDeviceListView.resize(m_DeviceList.size());
+	std::copy(m_DeviceList.begin(), m_DeviceList.end(), m_DpdkDeviceListView.begin());
```

Comment:
This can be simplified to `m_DpdkDeviceListView = m_DeviceList`, and usually this is slightly faster.

Reply:
I'm pretty sure that is not gonna work. `m_DeviceList` is a `PointerVector<T>` while `m_DpdkDeviceListView` is `std::vector<T>`.

I tested it with this code snippet:
```
pcpp::PointerVector<int> pVec;
std::vector<int*> vec;

pVec.pushBack(new int(1));
pVec.pushBack(new int(2));
pVec.pushBack(new int(3));

vec = pVec;
```

And it gives a no matching operator '=' compile time error.

Reply:
Ops, I didn't notice that the types are different. Sorry for the confusing.